task/build-image-index/0.1/build-image-index.yaml

```diff
@@ -74,15 +80,33 @@ spec:
 
       sed -i 's/^\s*short-name-mode\s*=\s*.*/short-name-mode = "disabled"/' /etc/containers/registries.conf
 
+      if [[ $# -ne 1 && "$ALWAYS_BUILD_INDEX" != "true" ]]; then
+        echo "Skipping image index generation while supplying multiple image inputs is unsupported."
+        exit 2
+      fi
+
       image_manifests=""
       buildah manifest create "$IMAGE"
       for i in $@
       do
         TOADD="$i"
-        if [[ $(echo $i | tr -cd ":" | wc -c) == 2 ]]; then
+        TOADD_REPOSITORY="$(echo "$i" | cut -d@ -f1)"
+        TOADD_DIGEST="$(echo "$i" | cut -d@ -f2)"
+        TOADD_TAG="$(echo "$IMAGE" | cut -d: -f2)"
```

Comment:
Shouldn't we be extracting the tag from `$i` rather than `$IMAGE`? Do we want to ignore the actual tag in `IMAGES` for some reason?

Note: this code wouldn't work as is, the tag extraction doesn't handle digests (and port numbers in the registry host, but that's a problem across almost every task in the pipeline)

Reply:
I added this tag logic because the e2e tests were failing since no tag was getting set. After looking at the pipeline definitions again, this is because I was passing the `IMAGE_REF` from the container build instead of `IMAGE_URL`. The result of this is that the IMAGE_URL that was returned ended up stripping off the tag.

For a matrix build, we need to pass `IMAGE_REF`, but for non-matrix builds, however, `$(tasks.build-container.results.IMAGE_URL)@$(tasks.build-container.results.IMAGE_DIGEST)` should be used to define the input `IMAGES` to allow the pass-through functionality.

Reply:
I'm not sure I understood that, but it would be great if the task worked in general, not only for some possible configurations of a pipeline